### PR TITLE
Enable code coverage pipeline with codecov

### PR DESCRIPTION
**Pull request by actione**
### Description

Target 1: Sync and display code coverage in PRs
This is the result of the tests conducted in my test PR after forking the project:
[ https://github.com/actione/cuda-quantum/pull/2 ](https://github.com/actione/cuda-quantum/pull/2 )
Target 2: Track branch coverage changes
[https://app.codecov.io/github/actione/cuda-quantum](https://app.codecov.io/github/actione/cuda-quantum)

Things to do to enable this feature:
-  Install the Codecov app for the cuda-quantum project. You can find it at: 
   https://github.com/apps/codecov

### CODE REVIEW

**Comment by actione** [comment]
I have read the Contributor License Agreement and I hereby accept the Terms.

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+coverage:
+  status:
+    project:
+      default:
+        target: 50%
+        threshold: 1%
+ignore:
+   - "tpl"
```
Comment (1tnguyen) [inline .github/codecov.yml:8]: Is it a typo here: `tpls`?

Comment (actione) [reply .github/codecov.yml]: thanks, fix in new commit.

**Comment by bmhowe23** [comment]
The `Load code coverage dependencies (amd64, llvm) / Caching` step took nearly 3 hours in the [CI run for this PR](https://github.com/NVIDIA/cuda-quantum/actions/runs/6012968397/job/16309405586). Do we expect it to always take that long, or was this just a one-time thing?

**Comment by bettinaheim** [comment]
Hi @actione, 
This is great. I am still following up on installing the codecov app, and I would like to make a pass to unify some of the build definitions that have been duplicated with this work. I need a bit more time but how to get back to this soon.

**Comment by actione** [comment]
Close this PR as I opened a new PR (#1510)  to track this.

**Pull request closed without merging**